src/gui/qgsrelationeditorwidget.h

```diff
@@ -255,7 +255,7 @@ class GUI_EXPORT QgsRelationEditorWidget : public QgsAbstractRelationEditorWidge
     QTreeWidget *mMultiEditTreeWidget = nullptr;
     QObjectUniquePtr<QgsMapToolDigitizeFeature> mMapToolDigitize;
     QButtonGroup *mViewModeButtonGroup = nullptr;
-    QgsVectorLayerSelectionManager *mFeatureSelectionMgr = nullptr;
+    QSharedPointer<QgsVectorLayerSelectionManager>mFeatureSelectionMgr = nullptr;
```

Comment:
Please use the std C++ smart pointer instead of the qt one here

Reply:
Could this be a unique_ptr?

Reply:
@m-kuhn @nyalldawson oh I realise now that there was no memory leak, i did not expect to find the delete in [qgsdualview](https://github.com/qgis/QGIS/blob/50fd3d42593d7ef167982c5d1d64fe1bef46147d/src/gui/attributetable/qgsdualview.cpp#L1267).
There was still one leak when the dual view get deleted